### ISSUE TITLE
xgboost-v7: train settlement tradable-EV target instead of outcome-only probability

### ISSUE THREAD

**Issue by phead198708** [phead198708/BiGan#99]
## Context

The xgboost-v6 paper-shadow work showed that high settlement confidence can arrive after the market has already priced the outcome. The executor can reject those cases with `p_side - ask >= threshold`, but this is currently a strategy-layer rule applied after the model emits `p_up` / `p_down`.

For v7, we should move part of that decision into the model: settlement should not only answer "will UP/DOWN win?", it should answer "is buying this settlement token at the current executable price positive EV?".

## Problem

Current settlement flow:

- Model predicts `p_up`, `p_down`, `p_neutral`.
- Executor chooses UP/DOWN from the probabilities.
- Executor applies freshness, confidence, and price/edge gates such as `p_side - ask >= settlement_edge_threshold`.

Observed issue in the 30-round v6 paper run:

- Most settlement signals were weak around `p_side ~= 0.52-0.55`.
- A few high-confidence signals (`p_side > 0.8`) arrived late in the round.
- Those high-confidence signals often had expensive asks (`0.80-0.90`), so tradable edge was thin even when the final outcome was correct.

This suggests the model is learning outcome direction better than tradable settlement EV.

## Goal

For xgboost-v7, add a settlement EV-oriented training target and reporting path so the model directly estimates whether a settlement entry is worth taking at the current price.

Candidate target family:

- `settlement_win`: final side wins at expiry.
- `settlement_entry_ask`: executable entry price from the feature row / top-of-book snapshot.
- `settlement_net_ev`: `payout - entry_ask - estimated_costs` or equivalent label.
- `settlement_tradable_edge`: positive when the entry price leaves enough expected value after costs/margin.
- Optional classifier: `should_enter_settlement` for rows satisfying positive EV after cost.

## Proposed Design

1. Build v7 training rows with both outcome and executable price context.
2. Label settlement entries by realized payout against entry ask, not just final UP/DOWN outcome.
3. Add round-age / seconds-to-expiry features and bucketed evaluation.
4. Report performance separately for:
   - outcome direction accuracy
   - tradable settlement hit rate
   - net PnL / EV after entry ask and costs
   - early/mid/late round buckets
5. Keep executor-side risk controls, but make the model output more directly aligned with tradeability.

## Acceptance Criteria

- Dataset builder emits v7 settlement EV labels and documents the formula.
- Training/evaluation reports include outcome accuracy and tradable-EV metrics separately.
- Replay report compares v6 outcome-probability gate vs v7 settlement-EV gate on the same historical windows.
- Round-age buckets are included, at minimum:
  - `0-180s`
  - `180-360s`
  - `360-540s`
  - `540s+`
- Executor integration plan is documented: what v7 fields it should consume, and which risk gates remain executor-only.
- No promotion decision is based only on raw outcome hit rate.

## Notes

This is intentionally v7 scope. The v6 executor can keep the ask-edge gate as a safety rule, but v7 should learn the tradable settlement opportunity directly.

**Comment by phead198708** [comment]
## Suggestion: make the **evaluation function** the core of v7 -- and split it into three distinct functions

Agree the crux is the evaluation function. The v6 paper-shadow failure mode (reproduced on the current 30-round run: **16 dedup signals / 11 rounds**, p_side buckets `0.50-0.55 x11`, `0.65-0.70 x1`, `0.85-0.90 x3`, `0.90-0.95 x1`) isn't just "the model is weak" -- it's that **what we trained on and what we selected on were different objectives, and both were scored on rows that can't be traded.** So before picking a target column, nail down three *separate* functions.

### 1. Three functions, not one

| Function | Role | v6 problem |
|---|---|---|
| Label / target `y_i` | what the head fits per row | outcome-only (UP/DOWN), price-agnostic |
| Training loss `L` | how the head fits `y_i` | unweighted mlogloss -> capacity spent on untradable mass |
| Selection / promotion metric `M` | scalar to pick gate + promote | PnL, but computed over **non-executable** rows |

v6 trained on outcome log-loss, then selected the gate on cost-adjusted PnL -- misaligned. And `M` was computed over every row including pre-round-start ones (offline **78 trades -> 1 executable** in the restricted replay). v7 should align all three around **realized, cost-aware, execution-eligible cashflow**, but specify each explicitly.

### 2. Don't regress `net_ev` directly -- regress the *market residual*

`settlement_net_ev = payout - entry_ask - cost` is the right thing to **report**, but a poor thing to **regress directly**: the ask already encodes the crowd probability. With `ask` as a feature, a direct EV regressor learns the near-efficient-market identity `EV ~= p_market - ask ~= 0` and collapses to ~0 everywhere -- exactly the flat 0.50-0.55 mass we see.

What makes money is **calibrated edge over the executable price**:

```
EV_buy_and_hold = p_true - entry_worst
              ~= (p_true - p_market_implied) + (p_market_implied - entry_worst)
                = edge - (cost / half-spread)
```

So the tradable quantity is `edge = p_true - p_market_implied`. Recommendation: **keep v6's calibrated p_side head (temperature scaling is good), feed executable price as context, and compute EV analytically at serving** instead of as an opaque regression target. Calibration stays independently testable (reliability curves), and the model is pushed to *beat* the price, not reproduce it. If you want a learned target, regress/classify the residual `e_i = win_i - market_implied_i`, not raw EV.

### 3. The selection / promotion metric `M` (the one to get right)

Compute `M` **only over execution-eligible rows** -- at/after round start, ask present, within `[min_seconds_to_expiry, max_seconds_to_expiry]`. This single filter is what stops the 78->1 illusion from returning.

Per eligible row, with calibrated `p_hat_i`, executable `a_i = ask_i + slip + fee`, decision `d_i = 1{ p_hat_i - a_i >= margin }`:

```
pnl_i = d_i * (win_i - a_i)        # buy-and-hold to settlement, win_i in {0,1}
```

Report all, gate on the **bold**:
- **net PnL after cost** and **PnL per round**
- **coverage** = rounds with >=1 eligible+admitted trade (v6's real failure: coverage ~= 0)
- hit rate among `d_i=1`, and **mean captured edge** `mean(p_hat - a)`
- a risk-adjusted form (PnL / deployed, or per-round Sharpe) for tie-breaks

Promotion gate = positive net PnL **AND** beats two baselines on the *same* windows: (1) the **v6 outcome-probability gate**, and (2) a **"buy at ask, no skill" (EV~=0)** baseline -- isolates skill from luck. Plus a calibration floor (ECE <= x) so `M` can't be gamed by a few overconfident tails.

### 4. Loss should be tradability-weighted

Weight each training row by its tradable stake / `|edge|` (0 for ineligible rows). The 30-round data is ~70% untradable 0.50-0.55 mass; unweighted log-loss optimizes mostly that mass. Weighting focuses model capacity where getting `p_hat` right changes money.

### 5. Time-bucketed evaluation is a *decision*, not just a report

Strong signals arrive late & expensive (asks 0.80-0.90). So bucket `M` by seconds-to-expiry (`0-180 / 180-360 / 360-540 / 540+`) **and** by entry-ask, and require **at least one populated (time x price) cell with positive realized EV and enough trades** for promotion -- never a global average dominated by early untradable rows. This also tells the executor *when* the edge actually lives.

### 6. Label formula to document (acceptance criterion)

```
entry_worst              = min(0.99, ask + buy_slippage + ask*fee_bps/1e4)   # reuse VolatilityLabelConfig policy
settlement_realized_pnl  = win - entry_worst                                 # win in {0,1}; buy-and-hold = ONE-WAY cost
settlement_tradable_edge = win - market_implied_prob                         # residual label, the learnable signal
should_enter_settlement  = 1 if settlement_realized_pnl > ev_margin else 0
EV_hat (serving)         = p_side_hat - entry_worst
```

Note: settlement is **buy-and-hold to expiry -> one-way entry cost**, not the volatility sleeve's round-trip `0.072`. Using round-trip cost here over-penalizes; keep the cost basis explicit and separate from the volatility head.

---

**TL;DR** -- Keep the calibrated probability head; add executable price as context; **make the model beat `market_implied`, not reproduce it.** Define the evaluation metric as **cost-aware realized PnL on execution-eligible rows only**, gated against both the v6 gate and a zero-skill baseline, and require a positive-EV (time x price) bucket rather than a global average.
